src/SymptomHistory/SelectSymptoms.tsx

```diff
@@ -1,124 +1,85 @@
-  const symptomLogEntryToEdit = logEntry ? JSON.parse(logEntry) : null
-  const isEditingLogEntry = symptomLogEntryToEdit !== null
+interface SelectSymptomsFormProps {
+  entryToEdit: SymptomEntry
+}
 
-  const initialSelectedSymptoms = symptomLogEntryToEdit?.symptoms || []
-  const [selectedSymptoms, setSelectedSymptoms] = useState<Symptom[]>(
-    initialSelectedSymptoms,
-  )
+export const SelectSymptomsForm: FunctionComponent<SelectSymptomsFormProps> = ({
+  entryToEdit,
+}) => {
+  useStatusBarEffect("dark-content", Colors.secondary10)
+  const { t } = useTranslation()
+  const navigation = useNavigation()
 
-  const handleOnPressSymptom = (selectedSymptom: Symptom) => {
-    if (selectedSymptoms.includes(selectedSymptom)) {
-      setSelectedSymptoms(selectedSymptoms.filter((s) => s !== selectedSymptom))
-    } else {
-      setSelectedSymptoms([...selectedSymptoms, selectedSymptom])
-    }
-  }
+  const { updateEntry } = useSymptomHistoryContext()
 
-  const handleOnPressDelete = async () => {
-    if (isEditingLogEntry) {
-      const result = await deleteLogEntry(symptomLogEntryToEdit.id)
-      if (result.kind === "success") {
-        showMessage({
-          message: t("symptom_checker.entry_deleted"),
-          ...Affordances.successFlashMessageOptions,
-        })
+  const initialSelectedSymptoms =
+    entryToEdit?.kind === "Symptoms"
+      ? entryToEdit.symptoms
+      : new Set<Symptom.Symptom>()
 
-        navigation.goBack()
-      } else {
-        showMessage({
-          message: t("symptom_checker.errors.deleting_symptom_log"),
-          ...Affordances.errorFlashMessageOptions,
-        })
-      }
-    }
-  }
+  const [selectedSymptoms, setSelectedSymptoms] = useState<
+    Set<Symptom.Symptom>
+  >(initialSelectedSymptoms)
 
-  const handleOnPressSave = async () => {
-    if (isEditingLogEntry) {
-      await updateSymptomLog()
+  const handleOnPressSymptom = (selectedSymptom: Symptom.Symptom) => {
+    const nextSymptoms = new Set(Array.from(selectedSymptoms))
+    if (nextSymptoms.has(selectedSymptom)) {
+      nextSymptoms.delete(selectedSymptom)
     } else {
-      await createNewSymptomLog()
+      nextSymptoms.add(selectedSymptom)
     }
+    setSelectedSymptoms(nextSymptoms)
   }
 
-  const createNewSymptomLog = async () => {
-    const result = await addLogEntry(selectedSymptoms)
-
-    if (result.kind === "success") {
-      completeOnPressSave()
-    } else {
-      showMessage({
-        message: t("symptom_checker.errors.adding_symptoms"),
-        ...Affordances.successFlashMessageOptions,
-      })
-    }
+  const handleOnPressNoSymptoms = () => {
+    setSelectedSymptoms(new Set<Symptom.Symptom>())
   }
 
-  const updateSymptomLog = async () => {
-    const result = await updateLogEntry({
-      ...symptomLogEntryToEdit,
-      symptoms: selectedSymptoms,
-    })
+  const handleOnPressSave = async () => {
+    const result = await updateEntry(
+      entryToEdit.date,
```

Comment:
We can probably refactor the `updateEntry` method to only take the `selectedSymptoms` and `entryToEdit` because the `date` is encapsulated in `entryToEdit`.